source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRContentStackView.mm

```diff
@@ -158,12 +198,53 @@ - (void)config:(nullable NSDictionary<NSString *, id> *)attributes
 
 - (CGSize)intrinsicContentSize
 {
-    return self.frame.size;
+    return self.combinedContentSize;
 }
 
 - (void)addArrangedSubview:(UIView *)view
 {
-    [self.stackView addArrangedSubview:view];
+    [_stackView addArrangedSubview:view];
+    // if view is hidden before observer is added, we update hidden view count manually
+    if (view.hidden) {
+        [_hiddenSubviews addObject:view];
+    }
+    [view addObserver:self forKeyPath:@"hidden" options:NSKeyValueObservingOptionNew | NSKeyValueObservingOptionOld context:&kToggleVisibilityContext];
+}
+
+// it's hard to know collection to know when filling the collection is done, this method
+// signals that the filling is done, and do the final visibility adjustment
+- (void)hideIfSubviewsAreAllHidden
+{
+    NSInteger count = [_hiddenSubviews count];
+    if (count and count == [_stackView.arrangedSubviews count]) {
+        self.hidden = YES;
+    }
+}
+
+- (void)observeValueForKeyPath:(NSString *)keyPath
```

Comment:
>observeValueForKeyPath [](start = 8, length = 22)

It's possible to receive notifications that you didn't register for, usually because some superclass also registered for notifications on this object. Generally you should check whether this notification is for you (in this case, by checking whether `context == &kToggleVisibilityContext`) and, if not, forward the notification on up to [super observeValueForKeyPath:...].